### ISSUE TITLE
Date Picker looks and behaves completely different with German locale (de)

### ISSUE THREAD

**Issue by hansprueller** [vmware/clarity#2772]
## Describe the bug

I am using the Date Picker component (v0.13) in my Angular 6 project, basically everything worked fine with default locale settings of the project. When I switch the default locale of the whole project to "de" (or "DE"), the Date Picker has a completely different look, not recogniseable as clarity component. also the usability/way how to use it is different. It seems like a completly new/other component was put on the page? 

## How to reproduce

1.  add Date Picker to the .html file of a component, e.g.:
 <input formControlName="fromDate" type="date" clrDate> 
2.  switch locale of the project to "de" by adding the following PROVIDER in app.module.ts:
{ provide: LOCALE_ID, useValue: 'de' }
3.  start app with "npm start" and navigate to the route showing the component
4.  See error
5. you can toggle locales with above PROVIDER between 'en' and 'de' and see the error go and come again

## Expected behavior

Date Picker should appear in same layout, design, look and feel and usability as with default locale (en), only with German locale format of the date string(dd.mm.yyyy instead of mm/dd/yyyy) and with German locale Month names etc. 

## Versions

**App**

* Angular: 6.0.6
* Node: 10.4.1
* Clarity: 0.13

**Device:**

* Type: MacBook Pro 13"
* OS: MacOS (high sierra)
* Browser Firefox Version 62.0.3 AND ALSO Chrome Version 70.0.3538.67

## Additional notes

please see screenshots attached.

![01_date_picker_default_en_locale_ok](https://user-images.githubusercontent.com/22539307/47362331-74747480-d6d4-11e8-8064-ecfba7531ac3.png)
![02_date_picker_with_de_locale_problem](https://user-images.githubusercontent.com/22539307/47362336-7807fb80-d6d4-11e8-88aa-22e9fcddeb1d.png)

**Comment by gnomeontherun** [comment]
Can you please provide a stackblitz as mentioned in the bug template to help us see the actual implementation you have? It is hard to know what code is running from a screenshot. https://stackblitz.com/edit/clarity-light-theme-v013

**Comment by Zyzle** [comment]
I think this is related to a similar issue we found running the date-picker for en-GB locale. It seems like the way angular requires locales to be loaded changed at some point, suggest the documentation is updated to add. 

The following seems to work
```typescript
import { registerLocaleData } from '@angular/common';
import localeDE from '@angular/common/locales/de';

registerLocaleData(localeDE, 'de');

@NgModule({
  providers: [
    {
      provide: LOCALE_ID,
      useValue: 'de'
    },
    // ...etc
```
Once this is done you should see the date-picker the way you expect:
<img width="288" alt="screenshot 2018-10-24 at 00 39 45" src="https://user-images.githubusercontent.com/3787524/47397167-e88b3880-d725-11e8-9baa-c235624d76cb.png">

**Comment by hansprueller** [comment]
Hi Zyzle,
perfect. That helped. You pointed it out very good: what we did is to follow the Clarity documentation step by step. So we thought we did everything correct. 

The missing pieces have been:

```
import { registerLocaleData } from '@angular/common';
import localeDE from '@angular/common/locales/de';

registerLocaleData(localeDE, 'de');
```

After doing that, everything works like a charm. So adding this to the Clarity documentation should do the trick and also help other people avoid falling over this stumbling stone.

For me, it is working now. 

Thanks.
